TA-3750: Setup command modules

ZgjimHaziri opened the pull request:
#### Description

Defined the needed command module directories and added the base `module.ts` file. Also added how will the codeowners look after the changes. Left them commented for now, and will uncomment as part of the tickets that migrate the commands of each area.

#### Checklist

- [x] I have self-reviewed this PR
- [x] I have tested the change and proved that it works in different scenarios
- [ ] I have updated docs if needed

lippisch committed:
Adding minor comments

lippisch committed:
Profile is now loaded as part of the context

lippisch committed:
Logger now writes logs into the config directory

lippisch committed:
fix for --debug

lippisch committed:
Migrated list command as an example implementation

lippisch committed:
modules can extend any root level command

lippisch committed:
entry point file can be of 3 variants now, supporting foldername as prefix.

lippisch committed:
Merge remote-tracking branch 'origin/master' into fli-contribution-model-refactoring

lippisch committed:
Modules are now structured by owning team, not command structure

lippisch committed:
Base API with some refactored error handling

lippisch committed:
minor tweaks to align with documentation

ZgjimHaziri committed:
TA-3749: Define foundation of the CLI

ZgjimHaziri committed:
TA-3749: Revert unnecessary change

ZgjimHaziri committed:
TA-3749: Revert unnecessary change

ZgjimHaziri committed:
TA-3749: Revert unnecessary change

ZgjimHaziri committed:
TA-3749: Revert unnecessary change

ZgjimHaziri committed:
TA-3749: yarn-lock update

ZgjimHaziri committed:
TA-3749: Rever package.json update

ZgjimHaziri committed:
TA-3749: Move code to top level

ZgjimHaziri committed:
Merge base branch

ZgjimHaziri committed:
TA-3749: Trigger build

ZgjimHaziri committed:
TA-3749: Comment test run

ZgjimHaziri committed:
TA-3750: Setup command modules

promeris commented on .github/CODEOWNERS:
```diff
@@ -1 +1,12 @@
 *   @celonis/navi @gencblakqorip9 @celonis/astro
+
+#/*                              @celonis/astro
+#/src                            @celonis/astro
+#/src/core                       @celonis/astro
+#/src/commands/pacman/           @celonis/astro
+#/src/commands/profile/          @celonis/astro
```
The first line `*` should cover these paths as well. 
```suggestion
```

ZgjimHaziri replied:
The `*` would define the set owners everywhere in the repo. We would not like to have ownership of the other team commands. Similar to cfg-ibc ownership.

promeris replied:
So if we forget to add codeowners for a new folder structure, we would rather have it orphan and not assign a fallback owner?

ZgjimHaziri replied:
Do you have another idea in mind?

promeris replied:
I was thinking that we should leave `* @celonis/astro`, and only specify overrides of ownership for specific folder structures. This would make sure all changes are covered from repository rules and require PRs for respective owners.

ZgjimHaziri replied:
Astro would get added as an unnecessary reviewer for each command change. I don't think we would want that.

Either way, we will have the `* @celonis/astro` until the refactoring is done, so we can decide then if we want to remove it.

ZgjimHaziri committed:
TA-3750: Merge base branch insto TA-3750-setup-modules

ZgjimHaziri committed:
TA-3750: Rename Pacman module to Config

ZgjimHaziri committed:
TA-3750: Remove non-Astro codeowners

ZgjimHaziri committed:
TA-3750: Update Pacman module description

ZgjimHaziri commented:
Updates:
- Renamed `pacman` module to `config`. Because Pacman is not a good description to features provided there.
- After discussing with @meritonhusaj and @GencBlakqoriP9, removed non-Astro codeowners during the refactoring.

ZgjimHaziri committed:
TA-3750: Fix merge conflicts

ksalihu commented:
> * Renamed `pacman` module to `config`. Because Pacman is not a good description to features provided there.

Are we making an exception to the owning area naming convention, or are we considering `config` to be the owning area?

ZgjimHaziri committed:
TA-3750: Rename config module

ZgjimHaziri commented:
> Are we making an exception to the owning area naming convention, or are we considering `config` to be the owning area?

Agree. Changed it to `configuration-management`, because it's more of an area.

jetakasabaqi commented on .github/CODEOWNERS:
```diff
@@ -1 +1,12 @@
+#/*                              @celonis/astro
+#/src                            @celonis/astro
+#/src/core                       @celonis/astro
+#/src/commands/pacman/           @celonis/astro
```
this should say configuration-managment now

ZgjimHaziri replied:
Updated.

ZgjimHaziri committed:
TA-3750: Rename config module in codeowners file
